### PR TITLE
bump controller-gen to v0.9.2

### PR DESCRIPTION
**Pull request by Huang-Wei**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

`hack/verify-crdgen.sh` leverages `controller-gen` to ensure CRD manifests are always verified. However, it was using an old version v0.6.2, and that is not quite compatible with the k8s version each release is targeting.

By looking at the `controller-gen`'s tags, v0.9.2 is targeting k8s v1.24, so this PR bumps controller-gen to v0.9.2, in hack/verify-crdgen.sh, and also update all CRD yamls to be compatible with controllger-gen v0.9.2.

Moreover, this PR formalizes the CRDs (ElasticQuota and PodGroup) to have the Status field served as a subresource (`/status`).

#### Which issue(s) this PR fixes:

Fixes #

### CODE REVIEW

**Comment by Huang-Wei** [comment]
@denkensk could you take a look at this PR?

I added a `// +kubebuilder:subresource:status` to PodGroup and ElasticQuota, which I believe is a best practice in operator development.

**Comment by fromanirh** [comment]
/retest
perhaps another flake?
```
 --- FAIL: TestElasticController (31.10s)
    elasticquota_controller_test.go:73: The CRD is ready to serve
    elasticquota_controller_test.go:96: Init scheduler success
    --- FAIL: TestElasticController/The_status_of_the_pod_changes_from_pending_to_running (10.32s)
        elasticquota_controller_test.go:285: The status of the pod changes from pending to running Waiting nowEQUsed error: timed out waiting for the condition
    --- FAIL: TestElasticController/The_status_of_the_pod_changes_from_running_to_others (10.31s)
        elasticquota_controller_test.go:285: The status of the pod changes from running to others Waiting nowEQUsed error: timed out waiting for the condition
    --- FAIL: TestElasticController/Different_resource_between_max_and_min (10.24s)
        elasticquota_controller_test.go:285: Different resource between max and min Waiting nowEQUsed error: timed out waiting for the condition 
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -37,3 +37,14 @@ Fixes #
 
 #### Special notes for your reviewer:
 
+#### Does this PR introduce a user-facing change?
```
Comment (fromanirh) [inline .github/PULL_REQUEST_TEMPLATE.md:40]: unrelated (important) change :)

**Review by fromanirh — COMMENTED** [review]
/lgtm

**Comment by denkensk** [comment]
I need to troubleshoot this error.

**Comment by Huang-Wei** [comment]
/hold

> I need to troubleshoot this error.

Do you think it's due to the CRD spec change?

---

I'm leaving the CI failure log here: https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_scheduler-plugins/459/pull-scheduler-plugins-integration-test-master/1603391796887425024

And let me rerun it to see if it's a flake.

/retest

**Comment by Huang-Wei** [comment]
@denkensk I think I know the root cause - when we standardize `status` as a subResource, when we patch the PodGroup/ElasticQuota object, we should explicitly specify its subResource path "/status" b/c for now status is not part of the regular API endpoint anymore.

Working on a fix.

**Comment by Huang-Wei** [comment]
The latest CI failure is due to NetworkAware plugin's flakiness.

**Comment by denkensk** [comment]
/lgtm
Thanks.

**Pull request merged**